lib/web/fetch/response.js

```diff
@@ -526,7 +527,7 @@ function fromInnerResponse (innerResponse, guard) {
   setHeadersGuard(response[kHeaders], guard)
 
   if (hasFinalizationRegistry && innerResponse.body?.stream) {
-    registry.register(response, innerResponse.body.stream)
+    registry.register(response, new WeakRef(innerResponse.body.stream))
```

Comment:
```js
registry.register(target, "some value");
```

> If target is reclaimed, your cleanup callback may be called at some point with the held value you provided for it ("some value" in the above). The held value can be any value you like: a primitive or an object, even undefined. **If the held value is an object, the registry keeps a strong reference to it (so it can pass it to your cleanup callback later).**

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/FinalizationRegistry

Reply:
if no one merges this before you can, adding this as a comment in the code itself would be appreciated